core/lib.rs

```diff
@@ -110,6 +129,24 @@ impl Connection {
         }
     }
 
+    pub fn update_pragma(&self, name: &String, value: i64) {
+        match name.as_str() {
+            "cache_size" => {
+                // update in-memory header
+                self.header.borrow_mut().default_cache_size = value
+                    .try_into()
+                    .expect(&format!("invalid value, too big for a i32 {}", value));
+
+                // update in disk
+                self.pager.write_database_header(&self.header.borrow());
```

Comment:
Oh wow, I did not expect you to do the write path too! Please note that we probably want this to go through a writeahead-log (WAL) for recovery purposes, but we can do that later.

Reply:
see above comment :(